src/client.ts

```diff
@@ -262,6 +266,103 @@ export class UXDClient {
+      controller.redeemableMintDecimals
+    );
+
+    return this.instruction.registerMaplePoolDepository(
+      mintingFeeInBps,
+      redeemingFeeInBps,
+      redeemableAmountUnderManagementCapBN,
+      {
+        accounts: {
+          authority,
+          payer: payer ?? authority,
+          controller: controller.pda,
+          depository: depository.pda,
+          depositoryCollateral: depository.depositoryCollateral,
+          collateralMint: depository.collateralMint,
+          maplePool: depository.maplePool,
+          maplePoolLocker: depository.maplePoolLocker,
+          mapleGlobals: depository.mapleGlobals,
+          mapleLender: depository.mapleLender,
+          mapleSharesMint: depository.mapleSharesMint,
+          mapleLockedShares: depository.mapleLockedShares,
+          mapleLenderShares: depository.mapleLenderShares,
+          systemProgram: SystemProgram.programId,
+          associatedTokenProgram: ASSOCIATED_TOKEN_PROGRAM_ID,
+          tokenProgram: TOKEN_PROGRAM_ID,
+          syrupProgram: depository.syrupProgramId,
+          rent: SYSVAR_RENT_PUBKEY,
+        },
+        options,
+      }
+    );
+  }
+
+  public createMintWithMaplePoolDepositoryInstruction(
+    controller: Controller,
+    depository: MaplePoolDepository,
+    authority: PublicKey,
+    collateralAmount: number,
+    options: ConfirmOptions,
+    payer?: PublicKey
+  ): TransactionInstruction {
+    const nativeCollateralAmount = uiToNative(
+      collateralAmount,
+      depository.collateralDecimals
+    );
+
+    const collateralMint = depository.collateralMint;
+    const redeemableMint = controller.redeemableMintPda;
+
+    const [[userCollateralATA], [userRedeemableATA]] = findMultipleATAAddSync(
+      authority,
+      [collateralMint, redeemableMint]
+    );
+
+    return this.instruction.mintWithMaplePoolDepository(
+      nativeCollateralAmount,
+      {
+        accounts: {
+          user: authority,
+          payer: payer ?? authority,
+          controller: controller.pda,
+          depository: depository.pda,
+          depositoryCollateral: depository.depositoryCollateral,
+          redeemableMint: redeemableMint,
+          userRedeemable: userRedeemableATA,
+          collateralMint: collateralMint,
+          userCollateral: userCollateralATA,
+          maplePool: depository.maplePool,
+          maplePoolLocker: depository.maplePoolLocker,
+          mapleGlobals: depository.mapleGlobals,
+          mapleLender: depository.mapleLender,
+          mapleSharesMint: depository.mapleSharesMint,
+          mapleLockedShares: depository.mapleLockedShares,
+          mapleLenderShares: depository.mapleLenderShares,
+          systemProgram: SystemProgram.programId,
+          associatedTokenProgram: ASSOCIATED_TOKEN_PROGRAM_ID,
+          tokenProgram: TOKEN_PROGRAM_ID,
+          syrupProgram: depository.syrupProgramId,
+          rent: SYSVAR_RENT_PUBKEY,
+        },
+        options,
+      }
+    );
```

Comment:
You can use property shorthand syntax when the name of the property is the same as the variable. Also can use destructuration to simplify? (this one is opiniated)

```suggestion
    const {
          depositoryCollateral,
          maplePool,
          maplePoolLocker,
          mapleGlobals,
          mapleLender,
          mapleSharesMint,
          mapleLockedShares,
          mapleLenderShares,
    } = depository;
   
    return this.instruction.mintWithMaplePoolDepository(
      nativeCollateralAmount,
      {
        accounts: {
          user: authority,
          payer: payer ?? authority,
          controller: controller.pda,
          depository: depository.pda,
          depositoryCollateral,
          redeemableMint,
          userRedeemable: userRedeemableATA,
          collateralMint,
          userCollateral: userCollateralATA,
          maplePool,
          maplePoolLocker,
          mapleGlobals,
          mapleLender,
          mapleSharesMint,
          mapleLockedShares,
          mapleLenderShares,
          systemProgram: SystemProgram.programId,
          associatedTokenProgram: ASSOCIATED_TOKEN_PROGRAM_ID,
          tokenProgram: TOKEN_PROGRAM_ID,
          syrupProgram: depository.syrupProgramId,
          rent: SYSVAR_RENT_PUBKEY,
        },
        options,
      }
    );
```